### PR TITLE
add/lys hub sidebar

### LINKED ISSUE #45115: [Launch Your Store] LYS hub - Sidebar
### Issue description

Figma: 0BWMerqRR1f468EwC7g5tg-fi-835_28896

Implement LYS hub - Sidebar 

Hub - Tasks

Implement only the following tasks if pending:

- Add your products
- Customize your store
- Get paid with WooPayments / other payment methods
- Collect sales taxes
- Set up shipping

Don’t include “Grow your business” even if it’s pending.

### Accept criteria

- Should contain sidebar
- Sidebar links to essential tasks not yet completed, and option to remove test orders

### PR DESCRIPTION
**Pull request by rjchow**
- **changed LYS task id to hyphens for consistency with the other tasks**
- **add: lys hub sidebar**

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/45115

### How to test the changes in this Pull Request:

This should be behind the Launch Your Store feature flag so it will be covered by a comprehensive call for testing when the feature is ready.

1. Enable LYS feature flag
2. Go to http://localhost:3000/wp-admin/admin.php?page=wc-admin&path=%2Flaunch-your-store
3. Inspect the design against the implementation and observe that
    - Tasks are correctly listed
    - Completed tasks are correctly styled
    - If all tasks are completed there should be the congratulatory text
    - If you have XState inspector set up, observe that the launch store button correctly triggers the event with the test order removal flag

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ public function get_id() {
 	 * @return string
 	 */
 	public function get_title() {
-		return __( 'Get your products shipped', 'woocommerce' );
+		return __( 'Select your shipping options', 'woocommerce' );
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php:48]: It looks like in the Figma file, the title of the task on LYS sidebar is `Select your shipping options`, while the title of the task on the setup task list is `Get your products shipped`.  🤔

Comment (rjchow) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php:48]: Good catch! Let me clarify

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀

File: plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/components/launch-store-hub.tsx [hunk 2]
```diff
@@ -1,40 +1,176 @@
...
+					) }
+				</ItemGroup>
+				<div className="edit-site-sidebar-navigation-screen-test-data__group-header">
+					<Heading level={ 2 }>
+						{ __( 'Test data', 'woocommerce' ) }
+					</Heading>
+				</div>
+				<ItemGroup className="edit-site-sidebar-navigation-screen-remove-test-data__group">
+					<ToggleControl
+						label={ __( 'Remove 20 test orders', 'woocommerce' ) }
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/components/launch-store-hub.tsx]: I assume `20` here is supposed to be dynamic, but perhaps this is going to be done in a follow-up?

I'd suggest to still use something like `sprintf` for crafting the string here

Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/components/launch-store-hub.tsx]: I also wonder now what happens if there's 0 test order? Should we remove the toggle altogether? 🤔

**Review by ilyasfoo — COMMENTED** [review]
It seems the homescreen tasklist is overridden when you click on "back" on one of the tasks that's opened from LYS screen, is this expected? 

1. Open Launch Your Store task
2. Click on "Add your products" 
3. Click on the back button on top of the task (not browser back)
4. Observe the different tasklist ([before](https://github.com/woocommerce/woocommerce/assets/3747241/6870fd1d-c08a-4d1a-b0d8-a73fe38ae518), [after](https://github.com/woocommerce/woocommerce/assets/3747241/c2d7f004-6bc6-4903-a582-f1d1a0379447))

It seems xstate5 is crashing when I try to inspect in a JN. I'm not sure if this is expected or not ([1](https://github.com/woocommerce/woocommerce/assets/3747241/7eea84f7-7d29-4038-b589-139405922ae8), [2](https://github.com/woocommerce/woocommerce/assets/3747241/5e2f5b88-1709-4b44-9550-a9862cd10e7f))

File: plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/components/launch-store-hub.tsx (hunk 2 above)
Comment (adrianduffell) [reply plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/components/launch-store-hub.tsx]: Ah sorry, we won't launch MVP with test order functionality. We'll follow-up with that shortly after.

Comment (rjchow) [reply plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/components/launch-store-hub.tsx]: > I assume 20 here is supposed to be dynamic, but perhaps this is going to be done in a follow-up?

I knew I forgot something... I filed it away to follow up but I didn't write it on my todo list. Might need to write a todo list app to help with that.

Thanks for catching this!!

> Ah sorry, we won't launch MVP with test order functionality. We'll follow-up with that shortly after

Ok I'll just make this section dependent on how many test orders were found, which we will default-populate with 0 for now

**Comment by rjchow** [comment]
> It seems the homescreen tasklist is overridden when you click on "back" on one of the tasks that's opened from LYS screen, is this expected?
> 
>     1. Open Launch Your Store task
> 
>     2. Click on "Add your products"
> 
>     3. Click on the back button on top of the task (not browser back)
> 
>     4. Observe the different tasklist ([before](https://github.com/woocommerce/woocommerce/assets/3747241/6870fd1d-c08a-4d1a-b0d8-a73fe38ae518), [after](https://github.com/woocommerce/woocommerce/assets/3747241/c2d7f004-6bc6-4903-a582-f1d1a0379447))

That's actually kinda weird, I didn't notice it because my back URL was else when I was testing it. Nice catch, I'll see what's up 🤔 

> It seems xstate5 is crashing when I try to inspect in a JN. I'm not sure if this is expected or not ([1](https://github.com/woocommerce/woocommerce/assets/3747241/7eea84f7-7d29-4038-b589-139405922ae8), [2](https://github.com/woocommerce/woocommerce/assets/3747241/5e2f5b88-1709-4b44-9550-a9862cd10e7f))

Might be some issues with build, let me look into this

File: plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/xstate.tsx [hunk 3]
```diff
@@ -17,22 +31,51 @@ import { updateQueryParams, createQueryParamsListener } from '../common';
...
+		'payments',
+		'shipping',
+		'tax',
+	];
+
+	const tasklist = await resolveSelect(
+		ONBOARDING_STORE_NAME
+	).getTaskListsByIds( [ 'setup' ] );
+	const visibleTasks: string[] = applyFilters(
+		'woocommerce_launch_your_store_tasklist_visible',
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/launch-your-store/hub/sidebar/xstate.tsx:69]: I recommend to add some docs for filters ([example](https://github.com/woocommerce/woocommerce/pull/44892/files#diff-cd9e87fb941aeb9dfec33993e6e06b5fa08023558c663baea4982f3ecf495c08R25-R30))

**Review by ilyasfoo — APPROVED** [review]
Tests well! I have another minor comment but should be fine as a follow-up 🚀

**Pull request merged**